### ISSUE TITLE
Can not use receptionist with BehaviorTestKit

### ISSUE THREAD

**Issue by hsn10** [akka/akka#29292]
I have lot of unit tests for actors written using BehaviorTestKit. Recently I added receptionist registration to actors and discovered that tests are starting to fail with.

```
java.lang.UnsupportedOperationException: ActorSystemStub cannot register extensions
```
It fails on simple registration message send:
```
context.system.receptionist ! Receptionist.Register(serviceKey, context.self)
```

I would like to have a mocked receptionist which accepts messages and BehaviorTestKit will have function to access its inbox.

**Comment by johanandren** [comment]
The BehaviorTestKit is meant for completely synchronous testing, as soon as you interact with other async things (or even sync things in extensions) outside of your control that falls apart.

What you could do is inject the `ActorRef[Receptionist.Command]` in your actor rather than look it up, that way you can probably use a probe in its place in your tests.

**Comment by patriknw** [comment]
I agree, and such ActorRef/Probe can be `TestInbox` and its `ref` when using BehaviorTestKit

**Comment by hsn10** [comment]
I think that sending registration message to receptionist should be inside scope of BehaviorTestKit. It is probably used a lot. Out of scope should be querying receptionist.

My proposal is to add mock probe receptionist into system and api receptionistInbox(): TestInbox[Receptionist.Command] into kit. It is simple to do.

**Comment by patriknw** [comment]
Sounds ok, please open a pull request

**Comment by ennru** [comment]
Fixed with #29294
